File: frontend/src/types/risk.ts
/**
 * Risk ve Bildirim TypeScript tipleri.
 * // Source: Doc 03 §3.7 + §3.9
 */

// ── Risk ──

export type RiskLevel = "low" | "moderate" | "high" | "critical";

export interface RiskStatus {
  overall_risk: RiskLevel;
  daily_loss: number;
  daily_loss_limit: number;
  open_positions: number;
  max_positions: number;
  rules_active: number;
  recent_events: RiskEventSummary[];
}

export interface RiskEventSummary {
  type: string;
  details: Record<string, unknown>;
  created_at: string;
}

export interface RiskRule {
  id: string;
  rule_type: string;
  value: Record<string, unknown>;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface RiskRuleUpdate {
  value?: Record<string, unknown>;
  is_active?: boolean;
}

export interface RiskEvent {
  id: string;
  user_id: string;
  event_type: string;
  rule_id: string | null;
  details: Record<string, unknown>;
  created_at: string;
}

// ── Notification ──

export interface Notification {
  id: string;
  user_id: string;
  type: string;
  title: string;
  body: string;
  channel: string;
  is_read: boolean;
  metadata_: Record<string, unknown>;
  sent_at: string;
  read_at: string | null;
}

File: backend/app/schemas/risk.py
# Source: Doc 03 §3.7 — Pydantic risk şemaları
"""Risk şemaları."""

from datetime import datetime
from decimal import Decimal
from enum import Enum
from uuid import UUID

from pydantic import BaseModel


class RiskLevel(str, Enum):
    LOW = "low"
    MODERATE = "moderate"
    HIGH = "high"
    CRITICAL = "critical"


class RiskStatusResponse(BaseModel):
    overall_risk: RiskLevel
    daily_loss: Decimal
    daily_loss_limit: Decimal
    open_positions: int
    max_positions: int
    rules_active: int
    recent_events: list[dict] = []


class RiskRuleResponse(BaseModel):
    id: UUID
    rule_type: str
    value: dict
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class RiskRuleUpdate(BaseModel):
    value: dict | None = None
    is_active: bool | None = None


class RiskEventResponse(BaseModel):
    """Risk olayı yanıt şeması."""
    id: UUID
    user_id: UUID
    event_type: str
    rule_id: UUID | None = None
    details: dict
    created_at: datetime

    model_config = {"from_attributes": True}
